### ISSUE TITLE
Incorrect argument order in AsyncFlateStreamHandler

### ISSUE THREAD

**Issue by ethanshry** [101arrowz/fflate#179]
**How to reproduce**
Install the library:
```bash
npm i --save fflate@0.8.0
```

Run the following code:

```ts
import { Decompress, gzipSync } from "fflate";

const d = new Decompress((err, chunk, final) => {
  console.log(err, chunk, final);
});

d.push(gzipSync(Buffer.from("Hello, fflate!")));
```
Declared type for the AsyncDecompress handler is:
```ts
export declare type AsyncFlateStreamHandler = (err: FlateError | null, data: Uint8Array, final: boolean) => void;
```
Expected log output: `null Uint8Array(14)... false`
Recieved log output: `Uint8Array(14)... false undefined`

**The problem**

As per the documentation, I would expect arguments to be in the order `err, chunk, final`, however it looks like arguments are instead `chunk, final, err`.

**Comment by 101arrowz** [comment]
`Decompress` is synchronous and accepts a `FlateStreamHandler`, see [these docs](https://github.com/101arrowz/fflate/blob/master/docs/classes/decompress.md#ondata). Errors are thrown by `push` and not sent to the callback at all. `AsyncDecompress` is a different class entirely that does have the expected arguments.

**Comment by ethanshry** [comment]
@101arrowz It makes sense to me that `Decompress` should use `FlateStreamHandler`, while `AsyncDecompress` would use `AsyncFlateStreamHandler`, however this is the class definition for `Decompress` I get when I goto definition in VSCode:
```ts
export declare class Decompress {
    private G;
    private I;
    private Z;
    private o;
    private s;
    private p;
    /**
     * The handler to call whenever data is available
     */
    ondata: FlateStreamHandler;
    /**
     * Creates a decompression stream
     * @param opts The decompression options
     * @param cb The callback to call whenever data is decompressed
     */
    constructor(opts: InflateStreamOptions, cb?: AsyncFlateStreamHandler);
    /**
     * Creates a decompression stream
     * @param cb The callback to call whenever data is decompressed
     */
    constructor(cb?: AsyncFlateStreamHandler);
    /**
     * Pushes a chunk to be decompressed
     * @param chunk The chunk to push
     * @param final Whether this is the last chunk
     */
    push(chunk: Uint8Array, final?: boolean): void;
}
```

**Comment by 101arrowz** [comment]
Ah I screwed up the types in the constructor. Will fix.

**Comment by 101arrowz** [comment]
Fixed and will release in v0.8.1.

**Comment by 101arrowz** [comment]
v0.8.1 should be out, let me know if there are any more issues!
